sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessEvaluationContext.java

```diff
@@ -368,23 +368,22 @@ public CounterSet getCounters() {
    * {@link InProcessPipelineOptions#isShutdownUnboundedProducersWithMaxWatermark()}.
    */
   public boolean isDone(AppliedPTransform<?, ?, ?> transform) {
-    boolean done = true;
+    if (!watermarkManager
+        .getWatermarks(transform)
+        .getOutputWatermark()
+        .equals(BoundedWindow.TIMESTAMP_MAX_VALUE)) {
```

Comment:
Due to risk of NPE, its generally better to do:
`CONSTANT.equals(expression)`
rather than
`expression.equals(CONSTANT)`